CRM/Pledge/Form/PledgeView.php

```diff
@@ -19,102 +19,93 @@
-
-    $title = $displayName .
-      ' - (' . ts('Pledged') . ' ' . CRM_Utils_Money::format($values['pledge_amount']) .
-      ' - ' . $values['financial_type'] . ')';
-
     // add Pledge to Recent Items
     CRM_Utils_Recent::add($title,
       $url,
-      $values['id'],
+      $this->getPledgeID(),
       'Pledge',
-      $values['contact_id'],
+      $this->getPledgeValue('contact_id'),
       NULL,
       $recentOther
     );
 
-    // Check if this is default domain contact CRM-10482
-    if (CRM_Contact_BAO_Contact::checkDomainContact($values['contact_id'])) {
-      $displayName .= ' (' . ts('default organization') . ')';
-    }
-    // omitting contactImage from title for now since the summary overlay css doesn't work outside of our crm-container
-    $this->setTitle(ts('View Pledge by') . ' ' . $displayName);
-
-    // do check for campaigns
-    $campaignId = $values['campaign_id'] ?? NULL;
-    if ($campaignId) {
-      $campaigns = CRM_Campaign_BAO_Campaign::getCampaigns($campaignId);
-      $values['campaign'] = $campaigns[$campaignId];
-    }
-
-    $this->assign($values);
+    $this->assign('campaign', $this->getPledgeValue('campaign_id:label'));
+    $currentInstallment = $this->getPledgeValue('amount') / $this->getPledgeValue('installments');
+    $this->assign('currentInstallment', $currentInstallment);
+    $this->assign('originalPledgeAmount', $currentInstallment === $this->getPledgeValue('original_installment_amount') ? NULL : $this->getPledgeValue('installments') * $this->getPledgeValue('original_installment_amount'));
+    $this->assign('start_date', $this->getPledgeValue('start_date'));
+    $this->assign('create_date', $this->getPledgeValue('create_date'));
+    $this->assign('end_date', $this->getPledgeValue('end_date'));
+    $this->assign('cancel_date', $this->getPledgeValue('cancel_date'));
+    $this->assign('is_test', $this->getPledgeValue('is_test'));
+    $this->assign('acknowledge_date', $this->getPledgeValue('acknowledge_date'));
+    $this->assign('pledge_status', $this->getPledgeValue('status_id:label'));
+    $this->assign('pledge_status_name', $this->getPledgeValue('status_id:name'));
+    $this->assign('amount', $this->getPledgeValue('amount'));
+    $this->assign('currency', $this->getPledgeValue('currency'));
+    $this->assign('initial_reminder_day', $this->getPledgeValue('initial_reminder_day'));
+    $this->assign('additional_reminder_day', $this->getPledgeValue('additional_reminder_day'));
+    $this->assign('max_reminders', $this->getPledgeValue('max_reminders'));
+    $this->assign('installments', $this->getPledgeValue('installments'));
+    $this->assign('original_installment_amount', $this->getPledgeValue('original_installment_amount'));
+    $this->assign('frequency_interval', $this->getPledgeValue('frequency_interval'));
+    $this->assign('frequency_day', $this->getPledgeValue('frequency_day'));
+    $this->assign('contribution_page', $this->getPledgeValue('contribution_page_id:label'));
```

Comment:
> @jitendrapurohit Should this be contribution_page_id.title?

That sounds correct to me. Can you do a PR?

Reply:
@colemanw doesn't the need to do that suggest we should declare a pseudoconstant? I feel like being able to refer to an entity by label is kinda one of those helpers that really makes code easier - perhaps in this case we might already know that the label field is title - but using :name & :label is so useful in general it's my go to rather than trying to figure out the label or name field for any given entity

Reply:
I've fixed to contribution_page_id.title - which works - but I find it really confusing that the handy syntax sometimes works & sometimes doesn't